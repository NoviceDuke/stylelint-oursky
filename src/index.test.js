var stylelint = require("stylelint");
var path = require("path");

function lint(code) {
  return stylelint.lint({
    code,
    formatter: "json",
    config: {
      plugins: [path.join(__dirname, "./index.js")],
      rules: {
        "oursky/flex": "error",
      },
    },
  }).then((result) => {
    return result.output;
  });
}

function check(code) {
  return lint(code).then(output => expect(output).toMatchSnapshot());
}

it("warns 1-value forms", async () => {
  return check("a{flex: 1;}");
});

it("warns 2-value forms", async () => {
  return check("a{flex: 1 0;}");
});

it("handles spaces", async () => {
  return check("a{flex: 1   0    ;}");
});

it("warns unitless flex-basis", async () => {
  return check("a{flex: 1   0  0  ;}");
});

it("warns unitless flex-basis of any number", async () => {
  return check("a{flex: 1   0  1  ;}");
});

it("passes well-formed flex", async () => {
  return check("a{flex: 1   0  0px  ;}");
});
