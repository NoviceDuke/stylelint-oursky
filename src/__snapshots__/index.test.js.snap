// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handles spaces 1`] = `"[{\\"source\\":\\"<input css 3>\\",\\"deprecations\\":[],\\"invalidOptionWarnings\\":[],\\"parseErrors\\":[],\\"errored\\":true,\\"warnings\\":[{\\"line\\":1,\\"column\\":3,\\"rule\\":\\"oursky/flex\\",\\"severity\\":\\"error\\",\\"text\\":\\"Expected 3-values form of flex: <flex-grow> <flex-shrink> <flex-basis>; (oursky/flex)\\"}]}]"`;

exports[`passes well-formed flex 1`] = `"[{\\"source\\":\\"<input css 6>\\",\\"deprecations\\":[],\\"invalidOptionWarnings\\":[],\\"parseErrors\\":[],\\"warnings\\":[]}]"`;

exports[`warns 1-value forms 1`] = `"[{\\"source\\":\\"<input css 1>\\",\\"deprecations\\":[],\\"invalidOptionWarnings\\":[],\\"parseErrors\\":[],\\"errored\\":true,\\"warnings\\":[{\\"line\\":1,\\"column\\":3,\\"rule\\":\\"oursky/flex\\",\\"severity\\":\\"error\\",\\"text\\":\\"Expected 3-values form of flex: <flex-grow> <flex-shrink> <flex-basis>; (oursky/flex)\\"}]}]"`;

exports[`warns 2-value forms 1`] = `"[{\\"source\\":\\"<input css 2>\\",\\"deprecations\\":[],\\"invalidOptionWarnings\\":[],\\"parseErrors\\":[],\\"errored\\":true,\\"warnings\\":[{\\"line\\":1,\\"column\\":3,\\"rule\\":\\"oursky/flex\\",\\"severity\\":\\"error\\",\\"text\\":\\"Expected 3-values form of flex: <flex-grow> <flex-shrink> <flex-basis>; (oursky/flex)\\"}]}]"`;

exports[`warns unitless flex-basis 1`] = `"[{\\"source\\":\\"<input css 4>\\",\\"deprecations\\":[],\\"invalidOptionWarnings\\":[],\\"parseErrors\\":[],\\"errored\\":true,\\"warnings\\":[{\\"line\\":1,\\"column\\":3,\\"rule\\":\\"oursky/flex\\",\\"severity\\":\\"error\\",\\"text\\":\\"flex-basis: 0 not allowed. Please include unit. (oursky/flex)\\"}]}]"`;

exports[`warns unitless flex-basis of any number 1`] = `"[{\\"source\\":\\"<input css 5>\\",\\"deprecations\\":[],\\"invalidOptionWarnings\\":[],\\"parseErrors\\":[],\\"errored\\":true,\\"warnings\\":[{\\"line\\":1,\\"column\\":3,\\"rule\\":\\"oursky/flex\\",\\"severity\\":\\"error\\",\\"text\\":\\"flex-basis: 1 not allowed. Please include unit. (oursky/flex)\\"}]}]"`;
